content/docs/rfcs/17/README.md

```diff
@@ -153,28 +162,37 @@ message WakuMessage {
 ## RateLimitProof
 
 The `proof` field is an array of 256 bytes and carries the zkSNARK proof as explained in the [Publishing process](##Publishing).
-The proof asserts that:
-1. The message publisher is the current member of the group i.e., her/his identity commitment key is part of the membership group Merkle tree with the root `merkle_root`.
-2. `share_x` and `share_y`  are correctly computed.
-3. The `nullifier` is constructed correctly.
 
-Other fields of the `RateLimitProof` message are the public inputs to the rln circuit and used for the generation of the `proof`.
+Other fields of the `RateLimitProof` message are the public inputs to the [rln circuit](/spec/32) and used for the generation of the `proof`.
 
-The `merkle_root` is an array of 32 bytes which holds the root of membership group Merkle tree at the time of publishing the message.
+The `merkle_root` is an array of 32 bytes in little-endian order which holds the root of membership group Merkle tree at the time of publishing the message.
 
-The `epoch` is an array of 32 bytes that represents the epoch in which the message is published.
-<!-- TODO epoch is going to change to a different type -->
+The `epoch` is an array of 32 bytes in little-endian order that represents the epoch in which the message is published.
```

Comment:
I expect `epoch` to be 8 bytes long or just a _bit_ more (Timestamp type).

Reply:
epoch is not of the timestamp type but is a counter and its type is dictated by the underlying circuit definition.
One important note about the epoch is that it corresponds to the external nullifier in the general rln construct, and conceptually can be any arbitrary value e.g., can represent a voting booth for which only one vote can be cast by each participant. As we are using rln for spam protection in a GossipSub network, we opted for the `epoch` definition which is smaller than 32 bytes, but in other rln applications, it may not be the case. I imagine the reason for it being 32bytes is to account for arbitrary values.